### ISSUE TITLE
Show previously installed apps

### ISSUE THREAD

**Issue by chadalen** [linuxappstore/linuxappstore-frontend#8]

**Comment by probonopd** [comment]
This would probably require some sort of login, which I would personally consider an anti-feature. I am not using commercial app stores because what I have installed on my machine is not anyone else's business. Privacy first!

**Comment by chadalen** [comment]
@probonopd The way I have this planned will not require you to login from the Linux App Store side. Your OS will ask you to enter your sudo password the same way gnome software does. The Linux App Store won't track any of that do your privacy is safe.

**Comment by chadalen** [comment]
@probonopd I do want privacy first. I could always store that information locally on your system? Or I could give you the option to choose how you want your data stored? I plan on having a settings page that will have a bunch of options so you can personalize the Linux app store how you want it.

**Comment by probonopd** [comment]
Keep in mind that with AppImages, users can store the files wherever they want to. I, for example, store my AppImages in a non-standard, read-only location. I doubt any system can know what is "installed" on my system, short of either scanning my whole system, or keeping track of what I have downloaded through some sort of login, which I would both find _very_ scaring.

**Comment by popey** [comment]
I don't believe it's productive to code for the very unique and bizarre use case @probonopd presents. While he is of course at liberty to setup his machine however he likes, the *vast* majority (millions) of users of Linux do *not* do this. snapd has an [api](https://github.com/snapcore/snapd/wiki/REST-API#get-v2snaps) to query what is installed. libflatpak has a similar [api](https://docs.flatpak.org/en/latest/libflatpak-api-reference.html#FlatpakInstalledRef) (note, their docs site has an invalid SSL cert). I agree it's probably sane to have this behind some kind of setting or option, so it can be turned off. But I think it makes sense to default on. An app store application which can't determine whether the user already has an application or not would make for an awful usability experience.

**Comment by probonopd** [comment]
> I don't believe it's productive to code for the very unique and bizarre use case @probonopd presents. 

This is not a "very unique and bizarre use case". AppImages have been designed from the beginning to be de-central, so that you can have as many AppImages in as many versions as you like anywhere in your filesystem. The AppImage approach is the opposite of a centralistic package manager or app store.

**Comment by popey** [comment]
I was referring to your comment "I, for example, store my AppImages in a non-standard, read-only location". That *is* very uncommon.

**Comment by probonopd** [comment]
...not for AppImage users ;-)
